diff --git a/scripts/package-build/linux-kernel/build-realtek-r8126.py b/scripts/package-build/linux-kernel/build-realtek-r8126.py
index 31bad0ce..f2144423 100755
--- a/scripts/package-build/linux-kernel/build-realtek-r8126.py
+++ b/scripts/package-build/linux-kernel/build-realtek-r8126.py
@@ -28,7 +28,7 @@ def add_depends(package_dir: str, package_name: str,
 defaults_file: str = Path('../../../data/defaults.toml').read_text()
 KERNEL_VER: str = toml_loads(defaults_file).get('kernel_version')
 KERNEL_FLAVOR: str = toml_loads(defaults_file).get('kernel_flavor')
-KERNEL_SRC: str = Path.cwd().as_posix() + '/linux'
+KERNEL_SRC: str = Path.cwd().as_posix() + '/build'
 # define variables
 PACKAGE_NAME: str = 'vyos-drivers-realtek-r8126'
 PACKAGE_VERSION: str = '10.016.00'
diff --git a/scripts/package-build/linux-kernel/build-realtek-r8152.py b/scripts/package-build/linux-kernel/build-realtek-r8152.py
index b8b3866d..b1163ac2 100755
--- a/scripts/package-build/linux-kernel/build-realtek-r8152.py
+++ b/scripts/package-build/linux-kernel/build-realtek-r8152.py
@@ -29,7 +29,7 @@ defaults_file: str = Path('../../../data/defaults.toml').read_text()
 architecture_file: str = Path('../../../data/architectures/amd64.toml').read_text()
 KERNEL_VER: str = toml_loads(defaults_file).get('kernel_version')
 KERNEL_FLAVOR: str = toml_loads(defaults_file).get('kernel_flavor')
-KERNEL_SRC: str = Path.cwd().as_posix() + '/linux'
+KERNEL_SRC: str = Path.cwd().as_posix() + '/build'
 # define variables
 PACKAGE_NAME: str = 'vyos-drivers-realtek-r8152'
 PACKAGE_VERSION: str = '2.21.4'
