diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
index 499f0c91c213..cf5b6bb6adbb 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dts
@@ -27,6 +27,7 @@ &gmac1 {
 	managed = "in-band-status";
 	phy-mode = "usxgmii";
 	sfp = <&sfp2>;
+	local-mac-address = [02 00 00 b3 14 42];
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
index 358ac1df35dd..14227c077027 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4.dtsi
@@ -166,6 +166,7 @@ &gmac2 {
 	managed = "in-band-status";
 	phy-mode = "usxgmii";
 	sfp = <&sfp1>;
+	local-mac-address = [02 00 00 b3 14 41];
 	status = "okay";
 };
 
@@ -181,7 +182,8 @@ &gsw_phy0_led0 {
 };
 
 &gsw_port0 {
-	label = "wan";
+	label = "eth3";
+	local-mac-address = [02 00 00 b3 14 43];
 };
 
 &gsw_phy1 {
@@ -196,7 +198,8 @@ &gsw_phy1_led0 {
 };
 
 &gsw_port1 {
-	label = "lan1";
+	label = "eth4";
+	local-mac-address = [02 00 00 b3 14 44];
 };
 
 &gsw_phy2 {
@@ -211,7 +214,8 @@ &gsw_phy2_led0 {
 };
 
 &gsw_port2 {
-	label = "lan2";
+	label = "eth5";
+	local-mac-address = [02 00 00 b3 14 45];
 };
 
 &gsw_phy3 {
@@ -226,7 +230,8 @@ &gsw_phy3_led0 {
 };
 
 &gsw_port3 {
-	label = "lan3";
+	label = "eth6";
+	local-mac-address = [02 00 00 b3 14 46];
 };
 
 &i2c0 {
